cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/DynamicPartitionerWithAvroTest.java

```diff
@@ -114,6 +114,9 @@ public void apply() throws IOException {
           Map<PartitionKey, PartitionDetail> partitions = new HashMap<>();
           for (PartitionDetail partition : pfs.getPartitions(null)) {
             partitions.put(partition.getPartitionKey(), partition);
+            // check that the mapreduce wrote the output partition metadata to all the output partitions
```

Comment:
Then you should add a test case. Otherwise, how will be sure that this code does not break in  the future?